### PR TITLE
Support download models from www.modelscope.cn

### PR DESCRIPTION
**Pull request by liuyhwangyh**
This PR implements download model from www.modelscope.cn, Compatible with the original version.

### CODE REVIEW

**Comment by simon-mo** [comment]
Hi @liuyhwangyh, can you help me understanding whether this can just be an example script in the example repository, without API changes? 

For example, can you just download the model from model hub ahead of time, and then load the model using existing API with model path?

**Comment by WoosukKwon** [comment]
A quick question: How popular is modelscope?

**Comment by simon-mo** [comment]
I believe it's the de-facto way to download models from China?

**Comment by liuyhwangyh** [comment]
add examples and document.

**Comment by wenmengzhou** [comment]
> Hi @liuyhwangyh, can you help me understanding whether this can just be an example script in the example repository, without API changes?
> 
> For example, can you just download the model from model hub ahead of time, and then load the model using existing API with model path?

Yes, downloading models and then passing the local path is feasible, but we hope users can use the vllm native interface to directly fetch models from ModelScope without having to write additional code for downloading logic, if possible.

If we switch to ModelScope through the configuration of environment variables without changing the API interface, thereby facilitating direct use by Chinese users, would this method be acceptable?

**Comment by simon-mo** [comment]
> If we switch to ModelScope through the configuration of environment variables without changing the API interface, thereby facilitating direct use by Chinese users, would this method be acceptable?

Yes. We would really like the least intrusive way possible here. The smaller amount of code change, the better. Thank you.

**Comment by liuyhwangyh** [comment]
> > If we switch to ModelScope through the configuration of environment variables without changing the API interface, thereby facilitating direct use by Chinese users, would this method be acceptable?
> 
> Yes. We would really like the least intrusive way possible here. The smaller amount of code change, the better. Thank you.

Hi @simon-mo @wenmengzhou ,According to the conversation, modified to control the downloading models from www.modelscope.cn through the environment variable FROM_MODELSCOPE.

File: examples/offline_inference.py [hunk 1]
```diff
@@ -10,13 +12,33 @@
...
-# Print the outputs.
-for output in outputs:
-    prompt = output.prompt
-    generated_text = output.outputs[0].text
-    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
+# Use model from www.modelscope.cn
+# FROM_MODELSCOPE=True python examples/offline_inference.py \
+# --model="damo/nlp_gpt2_text-generation_english-base" \
+# --revision="v1.0.0" --from_modelscope
+if __name__ == "__main__":
```
Comment (simon-mo) [inline examples/offline_inference.py:19]: for the example, I would like to keep it simple. The documentation is sufficient.

File: vllm/config.py [hunk 2]
```diff
@@ -76,7 +77,18 @@ def __init__(
         self.tokenizer_revision = tokenizer_revision
         self.quantization = quantization
 
-        self.hf_config = get_config(model, trust_remote_code, revision)
+        if os.environ.get("FROM_MODELSCOPE", False):
```
Comment (simon-mo) [inline vllm/config.py:80]: Let's call it `VLLM_USE_MODELSCOPE`

**Review by simon-mo — COMMENTED** [review]
Thanks for the quick update. Two main comments:
* Please revert the changes in examples. I think the documentation is sufficient.
* Please change the env var to `VLLM_USE_MODELSCOPE`. 

After that I think this PR is in good shape.

File: vllm/config.py (hunk 2 above)
Comment (liuyhwangyh) [reply vllm/config.py]: updated

File: examples/offline_inference.py (hunk 1 above)
Comment (liuyhwangyh) [reply examples/offline_inference.py]: reverted

File: docs/source/getting_started/quickstart.rst [hunk 3]
```diff
@@ -40,6 +40,14 @@ Initialize vLLM's engine for offline inference with the ``LLM`` class and the `O
 
     llm = LLM(model="facebook/opt-125m")
 
+Use model from www.modelscope.cn
+.. code-block:: shell
+    export VLLM_USE_MODELSCOPE=True
+
+.. code-block:: python
+
+    llm = LLM(model="damo/nlp_gpt2_text-generation_english-base", revision="v1.0.0")
```
Comment (yingdachen) [inline docs/source/getting_started/quickstart.rst:49]: I think "qwen/Qwen-7B-Chat" may be a better option to be used as an example.

File: docs/source/getting_started/quickstart.rst [hunk 4]
```diff
@@ -67,6 +75,15 @@ Start the server:
 
     $ python -m vllm.entrypoints.api_server
 
+Use model from www.modelscope.cn
+
+.. code-block:: console
+
+    $ VLLM_USE_MODELSCOPE=True python -m vllm.entrypoints.api_server \
+    $    --model="damo/nlp_gpt2_text-generation_english-base"
```
Comment (yingdachen) [inline docs/source/getting_started/quickstart.rst:83]: see other comment

File: vllm/config.py [hunk 5]
```diff
@@ -76,7 +77,18 @@ def __init__(
         self.tokenizer_revision = tokenizer_revision
         self.quantization = quantization
 
-        self.hf_config = get_config(model, trust_remote_code, revision)
+        if os.environ.get("VLLM_USE_MODELSCOPE", False):
+            # download model from modelscope
```
Comment (yingdachen) [inline vllm/config.py:81]: # download model from ModelScope hub, use on-demand import for compatability.

File: docs/source/getting_started/quickstart.rst (hunk 3 above)
Comment (liuyhwangyh) [reply docs/source/getting_started/quickstart.rst]: updated

File: docs/source/getting_started/quickstart.rst (hunk 4 above)
Comment (liuyhwangyh) [reply docs/source/getting_started/quickstart.rst]: done

File: vllm/config.py (hunk 5 above)
Comment (liuyhwangyh) [reply vllm/config.py]: done

**Review by simon-mo — APPROVED** [review]

**Pull request merged**